docs/_docs/02_features/keep-your-terragrunt-architecture-dry.md

```diff
@@ -409,3 +411,34 @@ terraform {
   source = "${include.env.locals.source_base_url}?ref=v0.2.0"
 }
 ```
+
+### Considerations for CI/CD Pipelines
+
+For infrastructure CI/CD pipelines, it is common to only want to run the workflow on the modules that were updated. For
+example, if you only changed the `terragrunt.hcl` configuration for the RDS database in the dev account, then you only
+want to run `plan` and `apply` on that module, not other components or other accounts.
+
+If you did not take advantage of `include` or `read_terragrunt_config`, then implementing this pipeline is
+straightforward: you can use `git diff` to collect all the files that changed, and for those `terragrunt.hcl` files that
+were updated, you can run `terragrunt plan` or `terragrunt apply` by passing in the updated file with
+`--terragrunt-config`.
+
+However, if you use `include` or `read_terragrunt_config`, then a single file change may need to be reflected on
+multiple files that were not touched at all in the commit. In our previous example, when a configuration is updated in
+the `_env/app.hcl` file, we need to apply the change to all the modules that `include` that common environment
+configuration.
+
+Terragrunt currently does not have any features for supporting this use case when `read_terragrunt_config` is
+used. However, for `include` blocks, you can use the
```

Comment:
Should we have something for `read_terragrunt_config`? Should `--terragrunt-modules-that-include` be something like `--terragrunt-modules-that-reference` instead, and support all possible ways to include or reference some other file?

Reply:
Ideally yes, but it's non-trivial to implement this using the tools that the `hcl` library gives us, since function calls can't be parsed out. That is, there is no way to say "give me all the calls to `read_terragrunt_config`," like there is a way to say "give me all the `include` blocks."

Reply:
I'm going to toss out an idea that may be dumb, but just in case: since we implement the `read_terragrunt_config` helper, could it maintain an internal cache (map) of (a) which files were read in and (b) which files were doing the reading? If so, could that cache be used to do the `--terragrunt-modules-that-include` style filtering?

Reply:
I think that would work, but I am a bit hesitant with that approach primarily because it is essentially relying on a global variable that gets updated with functions that have side effects. It feels... messy...

I filed https://github.com/gruntwork-io/terragrunt/issues/1910 so we don't lose track of it.

Reply:
Roger, thx!